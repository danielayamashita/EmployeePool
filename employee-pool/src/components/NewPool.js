import Nav from "./Nav";
import { useState } from "react";
import {handleCreatePool} from "../actions/pools"
import { connect} from "react-redux";
import { useNavigate } from "react-router-dom";
import LoginPage from "./LoginPage";

const NewPool = (props) => {

   const navigate = useNavigate();

  const [optionOneText, setOptionOneText] = useState("");
  const [optionTwoText, setOptionTwoText] = useState("");

  const handleChangeOptionOne = (e) => {
    e.preventDefault();
    setOptionOneText(e.target.value);
  };

  const handleChangeOptionTwo = (e) => {
    e.preventDefault();
    setOptionTwoText(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    
    props.dispatch(handleCreatePool(optionOneText,optionTwoText,props.authedUser));

    setOptionOneText("");
    setOptionTwoText("");

    // navigate(`/Dashboard/${props.authedUser}`)

  }

  return (
    <div>
      {props.authedUser ?
      <div className="container-new-pool">
        <form onSubmit={handleSubmit}>
          <h1>Would You Rather</h1>
          <h3>Create Your Own Pool</h3>

          <h5>First Option</h5>
          <input
            className="new-pool-form"
            type="text"
            placeholder="Option One"
            onChange={handleChangeOptionOne}
            value={optionOneText}
          />

          <h5>Second Option</h5>
          <input
            className="new-pool-form"
            type="text"
            placeholder="Option Two"
            onChange={handleChangeOptionTwo}
            value={optionTwoText}
          />
          <br />
          <button
            className="btn"
            type="submit"
            disabled={optionOneText === "" || optionTwoText === ""}
          >
            Submit
          </button>
        </form>
      </div>:
      <LoginPage/>
      }
    </div>
  );
};

const mapStateToProps = ({ authedUser}) => ({
    authedUser: authedUser,
});
export default connect(mapStateToProps)(NewPool);
