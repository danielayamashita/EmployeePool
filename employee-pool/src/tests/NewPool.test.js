


import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
// import { store } from './app/store';
import { MemoryRouter } from 'react-router'; 
import middleware from "../middleware"
import reducer from "../reducers";
import NewPool from "../components/NewPool"
import { createStore } from "redux";



const store = createStore(reducer,middleware);

store["authedUser"] = "mtsamis"

describe("NewPool", () => {
    it('renders NewPool component', () => {
    var component  = render(
        <MemoryRouter>
        <Provider store={store}>
        <NewPool />
        </Provider>
        </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
    });




});






