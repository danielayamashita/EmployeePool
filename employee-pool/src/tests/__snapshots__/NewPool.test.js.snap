// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPool renders NewPool component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="center"
        >
          <h1>
            Employee Pool
          </h1>
          <form>
            <h3>
              User
            </h3>
            <input
              data-testid="input-name"
              id="name"
              placeholder="User"
              type="User"
              value=""
            />
            <h3>
              Password:
            </h3>
            <input
              data-testid="input-password"
              id="name"
              placeholder="Password"
              type="text"
              value=""
            />
            <br />
            <button
              class="btn"
              data-testid="btn-submit"
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="center"
      >
        <h1>
          Employee Pool
        </h1>
        <form>
          <h3>
            User
          </h3>
          <input
            data-testid="input-name"
            id="name"
            placeholder="User"
            type="User"
            value=""
          />
          <h3>
            Password:
          </h3>
          <input
            data-testid="input-password"
            id="name"
            placeholder="Password"
            type="text"
            value=""
          />
          <br />
          <button
            class="btn"
            data-testid="btn-submit"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`isDataAPIAvailable renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="center"
        >
          <h1>
            Employee Pool
          </h1>
          <form>
            <h3>
              User
            </h3>
            <input
              data-testid="input-name"
              id="name"
              placeholder="User"
              type="User"
              value=""
            />
            <h3>
              Password:
            </h3>
            <input
              data-testid="input-password"
              id="name"
              placeholder="Password"
              type="text"
              value=""
            />
            <br />
            <button
              class="btn"
              data-testid="btn-submit"
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="center"
      >
        <h1>
          Employee Pool
        </h1>
        <form>
          <h3>
            User
          </h3>
          <input
            data-testid="input-name"
            id="name"
            placeholder="User"
            type="User"
            value=""
          />
          <h3>
            Password:
          </h3>
          <input
            data-testid="input-password"
            id="name"
            placeholder="Password"
            type="text"
            value=""
          />
          <br />
          <button
            class="btn"
            data-testid="btn-submit"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="center"
        >
          <h1>
            Employee Pool
          </h1>
          <form>
            <h3>
              User
            </h3>
            <input
              data-testid="input-name"
              id="name"
              placeholder="User"
              type="User"
              value=""
            />
            <h3>
              Password:
            </h3>
            <input
              data-testid="input-password"
              id="name"
              placeholder="Password"
              type="text"
              value=""
            />
            <br />
            <button
              class="btn"
              data-testid="btn-submit"
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="center"
      >
        <h1>
          Employee Pool
        </h1>
        <form>
          <h3>
            User
          </h3>
          <input
            data-testid="input-name"
            id="name"
            placeholder="User"
            type="User"
            value=""
          />
          <h3>
            Password:
          </h3>
          <input
            data-testid="input-password"
            id="name"
            placeholder="Password"
            type="text"
            value=""
          />
          <br />
          <button
            class="btn"
            data-testid="btn-submit"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
